pkg/allocator/cache.go

```diff
@@ -156,20 +161,57 @@ func (c *cache) OnDelete(id idpool.ID, key AllocatorKey) {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
-	c.onDeleteLocked(id, key)
+	c.onDeleteLocked(id, key, true)
+}
+
+func syncControllerName(id idpool.ID) string {
+	return fmt.Sprintf("sync-k8s-identity-%s", id)
```

Comment:
nit: could just use string concat per #19571 though this doesn't seem to be a likely hot path and k8s rpc will dominate any overhead difference

```suggestion
	return "sync-k8s-identity-"+id
```